### DISCUSSION TITLE
Does appium support beta versions of xcode?

### DISCUSSION THREAD

**Discussion by gselsidi** [appium/appium#15497] [Q&A]
How quickly does Appium support beta versions of xcode?

**Comment by KazuCocoa** [comment]
You can try out `appium@beta`
It already included some fixes https://github.com/appium/WebDriverAgent/commits/master
Then, please make sure your installed module versions.

**Reply by gselsidi** [reply]
Does this officially support Xcode beta?

**Reply by KazuCocoa** [reply]
Yes, if the "officially" means our fix/update WDA work. We've fixed issues that we figured out.
The Xcode is still in beta, so the Xcode itself also may have issues.

**Comment by jlipps** [comment]
Appium supports beta versions of Xcode as quickly as we can. There is no guarantee of support speed because every new version of Xcode presents its own challenges. We begin working on support as soon as the first developer beta is available, and usually release a new beta of Appium with support for the new Xcode as soon as there are any changes that have been made. Xcode stability changes over the course of the beta period. You're welcome to help test and update WebDriverAgent (our interface with Xcode) to make the process faster.

**Comment by cj-dovydas-razukas** [comment]
Is there any place where Appium Beta support can be tracked? Really interested in iOS 15 Beta support

**Reply by KazuCocoa** [reply]
So far, https://github.com/appium/appium-xcuitest-driver/issues/1312 and https://github.com/appium/WebDriverAgent/commits/master

No known remaining issues are there as Appium for Xcode 13 beta and iOS 15 beta by our regular test suites. Please use `appium@beta`.
For now, we reported an issue to Apple.

**Reply by diebold2000** [reply]
Pretty interesting because for now with Xcode 13 beta and iOS 15 beta on iPhone7 I was unable to run Appium tests with appium@beta. WDA and app are installed fine but then after I enabled UI Automation by fingerprint  - application just started and nothing is happening further.

`[WD Proxy] Got response with status 500: {"value":{"error":"unable to capture screen","message":"A screen capture was made impossible","traceback":""},"sessionId":"8FCD9F11-352B-41F4-A6AC-2403D2022213"}
[W3C] Matched W3C error code 'unable to capture screen' to UnableToCaptureScreen
[MJSONWP (720681d0)] Encountered internal error running command: UnableToCaptureScreen: A screen capture was made impossible
[MJSONWP (720681d0)]     at errorFromW3CJsonCode (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-base-driver/lib/protocol/errors.js:780:25)
![PXL_20210702_201908562](https://user-images.githubusercontent.com/7875241/124622956-361f4380-de84-11eb-8eb4-239253ade119.jpg)
![99A375C5-E62B-41EC-B019-A429C2B43909_1_201_a](https://user-images.githubusercontent.com/7875241/124623520-bcd42080-de84-11eb-9165-ca5e26468c32.jpeg)
`

**Reply by diebold2000** [reply]
By the way native XCUITests execution from Xcode is fine with the same configuration..

**Reply by KazuCocoa** [reply]
Do you have the system log?
If you have, creating an issue with log and appium log will help us https://github.com/appium/appium
Potentially Xcode does something internally for XCUITest via Xcode.

**Reply by KazuCocoa** [reply]
mm, I only had iPhone SE for iOS 15 beta 2. I just run a few steps with the latest `appium@beta`. It had no such behavior, so logs as https://github.com/appium/appium/discussions/15497#discussioncomment-972260 will help.
(Of course, "Automation Running" filter was on the view, but it did not affect screenshot)

**Comment by gselsidi** [comment]
gotcha thanks guys, unfortunately currently using appium through testproject.io so don't think they will add beta support of appium versions yet.

**Reply by KazuCocoa** [reply]
I see. It depends on cloud services if they support custom/beta appium in addition to regular appium versions.
